src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -188,7 +192,8 @@
       <_EmccLinkRsp>$(_WasmIntermediateOutputPath)emcc-link.rsp</_EmccLinkRsp>
 
       <EmccInitialHeapSize Condition="'$(EmccInitialHeapSize)' == ''">$(EmccTotalMemory)</EmccInitialHeapSize>
-      <EmccInitialHeapSize Condition="'$(EmccInitialHeapSize)' == ''">536870912</EmccInitialHeapSize>
+      <EmccInitialHeapSize Condition="'$(EmccInitialHeapSize)' == '' and '$(_WasmAssembliesTotalSize)' != '' and $(_WasmAssembliesTotalSize) > 16777216">$(_WasmAssembliesTotalSize)</EmccInitialHeapSize>
```

Comment:
If AssembliesTotalSize is the size of the assemblies without any padding, we'd want to do some sort of calculation here to add padding to it. I think msbuild can do math?

Reply:
Math is in the the C# Task and I could rename it to make that clearer. Suggestions for good name are welcome

Reply:
It's probably worth explicitly naming the task what it's for instead of the basis of its calculation - it's calculating the total size, then using it to predict or estimate the wasm heap size, so something like WasmEstimatedHeapSize maybe?

Reply:
I would go for something like `WasmCalculatedInitialHeapSize`